File: typescript/src/SimpleEnemy.ts
import { Target } from './Target';
import { Armor } from './Armor';
import { Buff } from './Buff';

export class SimpleEnemy extends Target {
    public constructor(private _armor: Armor, private _buffs: Array<Buff>) {
        super();
    }

    public get buffs(): Array<Buff> {
        return this._buffs;
    }

    public get armor(): Armor {
        return this._armor;
    }
}

File: go/simple_enemy.go
package codingdojo

type SimpleEnemy struct {
	// base Target
	armor Armor
	buffs []Buff
}

func makeSimpleEnemy(armor Armor, buffs []Buff) SimpleEnemy {
	return SimpleEnemy{armor, buffs}
}

func (this SimpleEnemy) getBuffs() []Buff {
	return this.buffs
}

func (this SimpleEnemy) getArmor() Armor {
	return this.armor
}
